tests/driver_hsc/main.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2020 Deutsches Zentrum für Luft- und Raumfahrt e.V. (DLR)
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the HSC pressure and temperature sensor
+ *
+ * @author      Quang Pham <phhr_quang@live.com>
+ *
+ * @}
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "errno.h"
+#include "string.h"
+#include "hsc.h"
+#include "hsc_params.h"
+#include "xtimer.h"
+
+#define SLEEP_USEC  (1UL * US_PER_SEC)
+
+int main(void)
+{
+    hsc_t dev;
+    int retval;
+
+    xtimer_usleep(SLEEP_USEC);
+
+    puts("HSC test application\n");
+
+    printf("+------------Initializing------------+\n");
+    do {
+        retval = hsc_init(&dev, &hsc_params[0]);
+        if (retval < 0) {
+            printf("Initialization error with code: %s\n", strerror(-retval));
```

Comment:
```suggestion
            if (IS_USED(MODULE_NEWLIB) || IS_USED(MODULE_PICOLIBC)) {
                printf("Initialization error with code: %s\n", strerror(-retval));
            }
            else {
                puts("Initialization failed");
            }
```

Apparently, `strerror()` is not provided by avrlibc. (Which makes sense for those, as they can return `const char *` pointers pointing to ROM, making this function quite expensive for AVR targets.)

The above change should result in `strerror()` only being used, when `newlib` or `picolibc` is used as C library. The `IS_USED()` macro is defined in `kernel_defines.h` if I recall correctly.